covid19-etl/etl/dsfsi.py

```diff
@@ -0,0 +1,418 @@
+    submitter_id = re.sub("[^a-z0-9-_]+", "-", submitter_id)
+    return submitter_id
+
+
+def format_demographic_submitter_id(subject_submitter_id):
+    return subject_submitter_id.replace("subject_", "demographic_")
+
+
+def normalize_current_status(status):
+    normalized = {
+        "?": None,
+        "Alive": "alive",
+        "Clinically Stable": "stable",
+        "critical condition": "critical",
+        "Critical condition": "critical",
+        "Dead": "deceased",
+        "Death": "deceased",
+        "Deceased": "deceased",
+        "deceased": "deceased",
+        "Died": "deceased",
+        "In recovery": "in recovery",
+        "In treatment": "in treatment",
+        "NA": None,
+        "Pos": "positive",
+        "Receiving Treatment": "in treatment",
+        "Recovered": "recovered",
+        "recovered": "recovered",
+        "Recovered (3/19/2020)": "recovered",
+        "Stable": "stable",
+        " Stable": "stable",
+        "stable condition": "stable",
+        "Tested Negative and hence is not considered as a case anymore": None,
+        "Under treatment": "in treatment",
+        "unstable": "unstable",
+    }
+
+    return normalized[status]
+
+
+def normalize_symptoms(symptoms):
+    normalized = {
+        "acute pneumonia": "pneumonia",
+        "Asymptomatic": "asymptomatic",
+        "breathing complications": "trouble breathing",
+        "chest pain": "persistent pain or pressure in the chest",
+        "cold": "fever* or feeling feverish/chills",
+        "Cold": "fever* or feeling feverish/chills",
+        "Cough Fever": "cough",
+        "cough": "cough",
+        "Cough": "cough",
+        "COVID-19 related symptoms": "COVID-19 related symptoms",
+        "fever and aches and headache": "fever or feeling feverish/chills",
+        "fever": "fever or feeling feverish/chills",
+        "Fever": "fever or feeling feverish/chills",
+        "Mild to moderate": "mild to moderate",
+        "muscle pain": "muscle or body aches",
+        "NA": "no data",
+        "rhinorrhea": "runny or stuffy nose",
+        "Severe": "severe",
+        "sneezing": "sneezing",
+        "tired": "fatigue (tiredness)",
+    }
+
+    list_of_symptoms = re.split(r", | and ", symptoms)
+
+    result = []
+    for symptom in list_of_symptoms:
+        if symptom.strip() in normalized:
+            result.append(normalized[symptom.strip()])
+    return result
+
+
+def normalize_date(d):
+    if not d:
+        return None
+
+    parsed_date = None
+    for fmt in ("%Y/%m/%d", "%Y-%m-%d", "%d-%m-%y", "%m/%d/%Y", "%d/%m/%Y", "%d-%b-%y"):
+        try:
+            parsed_date = datetime.datetime.strptime(d, fmt)
+        except ValueError:
+            pass
+
+    if not parsed_date:
+        return None
+
+    return parsed_date.strftime("%m/%d/%y")
+
+
+def normalize_condition(condition):
+    normalized = {
+        "NA": None,
+        "was treated for unspecified underlying condition": None,
+    }
```

Comment:
this doesn't look right

Reply:
@m0nhawk are we really using None for both values?

Reply:
I have removed that.